File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/question/entity/Questions.java
package com.yyjy.exam.entity.question.entity;

import com.yyjy.exam.entity.paper.entity.PaperQuestion;
import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;
import java.util.List;

@Entity
public interface Questions {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	long id();
	
	String title();
	
	QuestionType type();
	
	@JoinColumn(name = "category_id")
	@ManyToOne
	QuestionsCategories category();
	
	@IdView
	long categoryId();
	
	QuestionDifficulty difficulty();
	
	int score();
	
	@Nullable
	String analysis();
	
	@OneToMany(mappedBy = "question")
	List<PaperQuestion> paperQuestions();
	
	@OneToMany(mappedBy = "question")
	List<QuestionChoices> questionChoices();
	
	@OneToOne(mappedBy = "question")
	@Nullable
	QuestionAnswers questionAnswers();
	
	@Nullable
	LocalDateTime createTime();
	
	@Nullable
	LocalDateTime updateTime();
	
	@Column(name = "deleted_at")
	@LogicalDeleted("now")
	@Nullable
	LocalDateTime delFlag();
}

File: exam-system-server/entity/src/main/kotlin/com/yyjy/models/entity/Questions.kt
package com.yyjy.models.entity

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 题目表
 */
@Entity
interface Questions {

    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    val id: Long

    val title: String

    val type: String

    val multi: Boolean?

    @ManyToOne
    val category: com.yyjy.models.entity.Categories

    @IdView
    val categoryId: Long

    val difficulty: String

    val score: Int?

    val analysis: String?

//    @ManyToManyView(
//        prop = "paperQuestions",
//        deeperProp = "paper"
//    )
//    val papers: List<Paper>

    @OneToMany(mappedBy = "question")
    val paperQuestions: List<com.yyjy.models.entity.PaperQuestion>

    @OneToMany(mappedBy = "question")
    val questionChoices: List<com.yyjy.models.entity.QuestionChoices>

    @OneToOne(mappedBy = "question")
    val questionAnswers: com.yyjy.models.entity.QuestionAnswers?

    /**
     * 创建时间
     */
    val createTime: LocalDateTime?

    /**
     * 更新时间
     */
    val updateTime: LocalDateTime?

    /**
     * 删除时间
     */
    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    val delFlag: LocalDateTime?

}

